### ISSUE TITLE
Simplify how attachment works

### ISSUE THREAD

**Issue by huksley** [kovetskiy/mark#9]
Hi, thanks for your great tool!

## Context & explanation

I've tried to use markdown image syntax but it does not work.
Example:

```markdown
### This attachment list

![Sample image](191001_Homepage_hero_image_FI_EN.png)
```

Link created but attachment not uploaded.
I've tried adding meta tag
```bash
<!-- Attachment: 191001_Homepage_hero_image_FI_EN.png -->
```

Debug and tracing shows how image uploaded but image appears broken on the page.
Afterwards I edited page by hand.
Link HTML stayed the same.

After that I am trying to update from command line and it gives error:
```
2019-11-13 11:42:22 [TRACE] loaded macro "@\\{([^}]+)\\}"
                            ├─ template: ac:link:user
                            ├─ expr: @\{([^}]+)\}
                            └─ config: map[Name:${1}]
2019-11-13 11:42:22 [TRACE] loaded macro "WS-\\d+"
                            ├─ template: ac:jira:ticket
                            ├─ expr: WS-\d+
                            └─ config: map[Ticket:${0}]
2019-11-13 11:42:22 [TRACE] loaded macro ":done:"
                            ├─ template: ac:status
                            ├─ expr: :done:
                            └─ config: map[Color:Green Title:DONE]
2019-11-13 11:42:22 [TRACE] loaded macro ":todo:"
                            ├─ template: ac:status
                            ├─ expr: :todo:
                            └─ config: map[Color:Blue Title:TODO]
2019-11-13 11:42:22 [DEBUG] resolving page path: ??? > Test
2019-11-13 11:42:23  [INFO] page will be stored under path: Solutions Architects Space > Test
2019-11-13 11:42:23  [INFO] updating attachment: "191001_Homepage_hero_image_FI_EN.png"
2019-11-13 11:42:26 [FATAL] unable to create/update attachments
                            └─ unable to update attachment "191001_Homepage_hero_image_FI_EN.png"
                               └─ Confluence REST API for creating attachments returned 0 json objects, expected at least 1
```

I am using Confluence Cloud.

## What happened

Images referenced in markdown, relative to the file are not uploaded without specific meta header Attachment. When specifying Attachment images are uploaded somewhere, but link still broken on the Confluence page.

## What I expect

Markdown image reference links works.
Both inline and locally referenced images should be uploaded and linked properly.

I used markdown format as described here:
https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#images

**Comment by kovetskiy** [comment]
Hi! I'm glad that you've found this tool useful.

It's quite surprising that the documentation doesn't cover how attachments should be linked in the docs. 

Here is how I do it right now:
```
registries.md
4:<!-- Attachment: assets/registries.png -->
50:![Image Flow](attachment://assets/registries.png)

clusters-overview.md
4:<!-- Attachment: assets/live-kubernetes.png -->
75:<img src="attachment://assets/live-kubernetes.png" width="800px" />
```

As you can see, you just need to specify the attachment upload section (you have it already) and mention the attachment with special `attachment://` protocol (mark will just replace it with full attachment URL received after uploading)

**Comment by huksley** [comment]
Hmm, yes, specifying `attachment://` prefix helps fixing url to the image.
What is the rationale for this? It would be great if links will be compatible with GitHub and page could be rendered both at github and confluence.

Also why `<!-- Attachment: assets/registries.png -->`
are required. I cant put it near the image tag (only at the top) so you need to go extra mile to add an image.

Overally IMO image support a little bit crude right now I would say. It would be great if it could be improved!

**Comment by kovetskiy** [comment]
Yes, I completely agree with you on this point and I don't even remember
why there is that virtual URI scheme, but I'd be glad to merge a PR with
improvements.

**Comment by csoutherland** [comment]
Maybe this will be sufficient? https://github.com/kovetskiy/mark/pull/22

**Comment by kovetskiy** [comment]
Should be fixed with 3.0 thanks for @csoutherland PRs.
